branch.go

```diff
@@ -0,0 +1,196 @@
+package flow
+
+import (
+	"context"
+	"sync"
+)
+
+// BranchCheckFunc checks the target and returns true if the branch should be taken.
+type BranchCheckFunc[T Steper] func(context.Context, T) (bool, error)
+
+// If adds a conditional branch to the workflow.
+//
+//	If(someStep, func(ctx context.Context, someStep *SomeStep) (bool, error) {
+//		// branch condition here, true -> Then, false -> Else.
+//		// if error is returned, then fail the selected branch step.
+//	}).
+//	Then(thenStep).
+//	Else(elseStep)
+func If[T Steper](target T, check BranchCheckFunc[T]) *IfBranch[T] {
+	return &IfBranch[T]{Target: target, BranchCheck: BranchCheck[T]{Check: check}}
+}
+
+// IfBranch adds target step, then and else step to workflow,
+// and check the target step and determine which branch to go.
+type IfBranch[T Steper] struct {
+	Target      T // the target to check
+	BranchCheck BranchCheck[T]
+	ThenStep    []Steper
+	ElseStep    []Steper
+	Cond        Condition // Cond is the When condition for both ThenStep and ElseStep, not target Step!
```

Comment:
What if the target step is skipped? How does it work?

Reply:
if the target step is skipped, then it's the `Cond` taking effect.
ThenStep and ElseStep depend on target step, and they share the `Cond`, if the `Cond` is `AllSucceeded`,
ThenStep and ElseStep will be skipped if the target skipped.
Users have this option to change the `Cond`, for instance, if someone 
```
If(target, ...).
    When(flow.AnyFailed),
```
ThenStep and ElseStep will only be evaluated when target failed.

Reply:
Sounds good